gtda/diagrams/distance.py

```diff
@@ -192,8 +192,9 @@ def fit(self, X, y=None):
             weight_function = self.effective_metric_params_.get(
                 'weight_function', None
                 )
-            if weight_function is None:
-                self.effective_metric_params_['weight_function'] = np.ones_like
+            weight_function = \
+                np.ones_like if weight_function is None else weight_function
```

Comment:
Isn't this equivalent to `weight_function = self.effective_metric_params_.get( 'weight_function', np.ones_like)`?

Reply:
Or, `self.effective_metric_params_.setdefault(np.ones_like)`?

Reply:
Not quite I think, because I think we'd like to allow the user to place `None` and yet we have to pass `np.ones_like` to `_parallel_amplitude` etc.

Reply:
But I might be wrong again!

Reply:
you're right!